Maniacs Feature - "Skip Dialogs" from Fatal Mix.

jetrotal opened the pull request:
Hold ESC to skip all dialogs when in Debug Mode.

Things to consider:
 
- Need an alternative to `strlen(text_index) < 3`. Since the textboxes can stop while being bigger than 3 characters, 
if the com.string has more than 4 lines of text.
- We need a flag that works as an alternative to `Player::debug_flag`, so people can use it in their games 
(both shift and this skip should be activated?)

jetrotal committed:
Skip Dialogs - Basic Implementation

Ghabry commented:
FYI: There is already "message appears without delay" when holding shift. So you can already hold shift and hammer decision for the same result.

jetrotal commented:
yeah, what I did was similar to #2490

jetrotal commented:
After a second look, they want something similar do Maniac's Fatal Mix as a flag:
![image](https://github.com/EasyRPG/Player/assets/45118493/822f2b19-9311-4ddc-85c2-a517959c1906)


Weird part, is that it's attached to Right Shift. That could be a problem

jetrotal committed:
Organizing Maniac's SetGameOption

That will help me seeing how to implement the alternative skip dialogs.

Update game_interpreter.cpp

Ghabry commented:
Do you want this auto-enter behaviour available in general or just when this fatal mix skip option is enabled?

jetrotal commented:
Oh boy, lemme try to remember how I set this up...

If I remember correctly, my code was very similar to the fatal mix behavior.
But I found it useful on vanilla games as well. 
It was a faster skip withtout hammering enter.

Right now, maybe we should stick it to the fatal mix?

Maybe in the future, would be interesting to have a way of accessing maniacs commands even with it disabled...?

Ghabry commented:
Considering it is a debug feature it could be an option in the settings whether you want "Instant text" with Shift or "Instant auto enter text" with Shift. :thinking:

jetrotal commented:
> Considering it is a debug feature it could be an option in the settings whether you want "Instant text" with Shift or "Instant auto enter text" with Shift. 🤔

Sure, it could also come with a setting

Ghabry commented:
There is another PR #3487 which also implements this and is more feature complete
